src/java.base/share/classes/java/lang/String.java

```diff
@@ -524,13 +533,11 @@ public String(int[] codePoints, int offset, int count) {
             return;
         }
         if (COMPACT_STRINGS) {
-            byte[] val = StringLatin1.toBytes(codePoints, offset, count);
-            if (val != null) {
-                this.coder = LATIN1;
-                this.value = val;
-                return;
-            }
-            initCompressionFlag();
+            byte[] val = StringUTF16.compress(codePoints, offset, count);
+            this.coder = StringUTF16.coderFromArrayLen(val, count);
+            this.value = val;
+            initCompressionFlag();	
```

Comment:
Calls to `initCompressionFlag()` should only occur if `coder == UTF16` (signalling the existence of non-compressed strings).

Reply:
Fixed